connectors/llm/interface.py

```diff
@@ -38,28 +43,55 @@ def ask(self, system_prompt, previous_messages, question, agent_id, stream):
                 if "title" in metadata:
                     title = metadata["title"]
                     context_text += f", document title: '{title}'"
-                context_text += ">>\n\n" f"{page_content}\n\n" f"<<Search result {i+1} END>>\n"
+                context_text += (">>\n\n" f"{page_content}\n\n" f"<<Search result {i+1} END>>\n")
 
         prompt = ChatPromptTemplate.from_template(cfg.USER_PROMPT_TEMPLATE)
         prompt_params = {"context": context_text, "question": question}
         log.debug("search result: %s", context_text)
 
+        text_splitter = MistralTokenizer.v3(is_tekken=True)
+
         # Adding system prompt and memory
         msg_list = []
-        msg_list.append(SystemMessage(content=system_prompt or cfg.DEFAULT_SYSTEM_PROMPT))
+        msg_list.append(
+            LangchainSystemMessage(content=system_prompt or cfg.DEFAULT_SYSTEM_PROMPT)
+        )
+
+        prompt_content = MistralSystemMessage(content=msg_list[0].content)
+        total_tokens = len(text_splitter.encode_chat_completion(ChatCompletionRequest(messages=[prompt_content])).tokens)
         if previous_messages:
-            for msg in previous_messages:
+            # Reverse list so most recent msgs are in context
+            for msg in reversed(previous_messages):
+                # Declare before if to avoid 'unbound' errors
+                m = HumanMessage(content=f"")
+                token_list = 0
+
                 if msg["sender"] == "human":
-                    msg_list.append(HumanMessage(content=f"[INST] {msg['text']} [/INST]"))
+                    token_list = len(text_splitter.encode_chat_completion(ChatCompletionRequest(messages=[UserMessage(content=msg["text"])])).tokens)
```

Comment:
Would it be OK if we change the name of this variable? I was expecting `token_list` to be a list type ... but it is just an integer right?